packages/sanity/src/core/studio/manifest/registerStudioManifest.tsx

```diff
@@ -0,0 +1,130 @@
+import {type SanityClient} from '@sanity/client'
+import {firstValueFrom} from 'rxjs'
+
+import {type Source, type WorkspaceSummary} from '../../config/types'
+import {type UserApplication} from '../../store/userApplications'
+import {resolveIcon} from './icon'
+
+/**
+ * Workspace configuration for the Studio manifest.
+ * @internal
+ */
+interface WorkspaceManifest {
+  name: string
+  projectId: string
+  dataset: string
+  schemaDescriptorId: string
+  basePath?: string
+  title?: string
+  subtitle?: string
+  icon?: string
+  mediaLibraryId?: string
+  apiHost?: string
+}
```

Comment:
Not for now, but long-term I think it would be good to understand where (and if) we want to share these types, because e.g. Dashboard makes use of them and has their own definitions https://github.com/sanity-io/ada/blob/main/packages/message-protocol/src/types.ts#L17